content/browser/renderer_host/brave_navigation_controller_impl.cc

```diff
@@ -0,0 +1,44 @@
+/* Copyright (c) 2023 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/content/browser/renderer_host/brave_navigation_controller_impl.h"
+
+#include "brave/components/constants/webui_url_constants.h"
+#include "build/build_config.h"
+#include "content/public/common/url_constants.h"
+
+namespace content {
+
+BraveNavigationControllerImpl::BraveNavigationControllerImpl(
+    BrowserContext* browser_context,
+    FrameTree& frame_tree,
+    NavigationControllerDelegate* delegate)
+    : NavigationControllerImpl(browser_context, frame_tree, delegate) {}
+
+BraveNavigationControllerImpl::~BraveNavigationControllerImpl() {}
+
+base::WeakPtr<NavigationHandle>
+BraveNavigationControllerImpl::LoadURLWithParams(const LoadURLParams& params) {
+  if (params.url.SchemeIs(content::kChromeUIUntrustedScheme) &&
+      params.url.host() == kChatUIHost) {
```

Comment:
Should there be a browser tests for this?

Reply:
Yes, it would be better, fixed at 
https://github.com/brave/brave-core/pull/20996/commits/29c81aed8b7bf633dcae5b9fab10aa04c4a46105